cli/src/config.rs

```diff
@@ -1373,36 +1502,37 @@ mod tests {
     }
 
     enum Want {
-        None,
         New(&'static str),
-        ExistingAndNew(&'static str),
+        Existing(&'static str),
     }
 
     struct TestCase {
         files: Vec<&'static str>,
         env: UnresolvedConfigEnv,
-        want: Want,
+        wants: Vec<Want>,
     }
 
     impl TestCase {
-        fn resolve(&self, root: &Path) -> Result<ConfigEnv, ConfigEnvError> {
+        fn resolve(&self, root: &Path) -> ConfigEnv {
             let home_dir = self.env.home_dir.as_ref().map(|p| root.join(p));
             let env = UnresolvedConfigEnv {
                 config_dir: self.env.config_dir.as_ref().map(|p| root.join(p)),
                 home_dir: home_dir.clone(),
-                jj_config: self
-                    .env
-                    .jj_config
-                    .as_ref()
-                    .map(|p| root.join(p).to_str().unwrap().to_string()),
+                jj_config: self.env.jj_config.as_ref().map(|p| {
+                    std::env::split_paths(p)
+                        .filter(|path| !path.as_os_str().is_empty())
+                        .map(|p| root.join(p).to_str().unwrap().to_string())
```

Comment:
Empty paths shouldn't be filtered out here because we need to test the actual implementation.